xs/src/libslic3r/Model.cpp

```diff
@@ -20,6 +20,9 @@ Model::Model(const Model &other)
     this->objects.reserve(other.objects.size());
     for (ModelObjectPtrs::const_iterator i = other.objects.begin(); i != other.objects.end(); ++i)
         this->add_object(**i, true);
+
+    // copy metadata
+    this->metadata = other.metadata;
```

Comment:
@lordofhyphens  The metadata is a map. If it's empty I do nothing.
The part number is specific to the model object and I initialize it by -1 if it's not found.